src/main/java/io/vertx/core/http/impl/Http1xClientConnection.java

```diff
@@ -897,8 +900,19 @@ public void createStream(ContextInternal context, HttpClientRequestImpl req, Pro
     }
   }
 
+  @Override
+  public boolean isValid() {
+    return expirationTimestamp > 0 && System.currentTimeMillis() <= expirationTimestamp;
+  }
+
   private void recycle() {
-    long expiration = keepAliveTimeout == 0 ? 0L : System.currentTimeMillis() + keepAliveTimeout * 1000;
-    listener.onRecycle(expiration);
+    if(!options.isKeepAliveTTLEnabled())
```

Comment:
this should actually be moved to `isValid` and when initialTimestamp > keepAliveTTL then return false